test/envtest/gpudeviceplugin_controller_test.go

```diff
@@ -58,16 +62,53 @@ var _ = Describe("GpuDevicePlugin Controller", func() {
 				return len(fetched.Status.ControlledDaemonSet.UID) > 0
 			}, timeout, interval).Should(BeTrue())
 
-			By("updating image name successfully")
+			By("DaemonSet created successfully")
+			ds := &apps.DaemonSet{}
+			_ = k8sClient.Get(context.Background(), types.NamespacedName{Namespace: ns, Name: "intel-" + "gpu" + "-plugin"}, ds)
+			Expect(ds.Spec.Template.Spec.NodeSelector).To(Equal(fetched.Spec.NodeSelector))
+
+			for _, dsInitContainer := range ds.Spec.Template.Spec.InitContainers {
+				Expect(dsInitContainer.Image).To(Equal(spec.InitImage))
+			}
```

Comment:
so if currently expect there to be one initcontainer and its name to be known we could

```suggestion
			Expect(ds.Spec.Template.Spec.InitContainers).To(HaveLen(1))
			Expect(ds.Spec.Template.Spec.InitContainers[0].Image).To(Equal(spec.InitImage))}
```

the logic would have to change if new InitContainers are added later but this works today

Reply:
Sorry, I already made the changes, but didn't push the commit.